### ISSUE TITLE
borg init does not work [WinError 10038]

### ISSUE THREAD

**Issue by henfri** [borgbackup/borg#4811]
Hello,

@jrast 
Under Windows borg init does not work:
```
C:\Users\henfri\AppData\Roaming\Python\Python37\Scripts>borg init --encryption=repokey "C:\Users\henfri\Desktop\temp\Borg\"
Local Exception
Traceback (most recent call last):
  File "C:\Users\henfri\AppData\Roaming\Python\Python37\site-packages\borg\archiver.py", line 4466, in main
    exit_code = archiver.run(args)
  File "C:\Users\henfri\AppData\Roaming\Python\Python37\site-packages\borg\archiver.py", line 4397, in run
    return set_ec(func(args))
  File "C:\Users\henfri\AppData\Roaming\Python\Python37\site-packages\borg\archiver.py", line 148, in wrapper
    make_parent_dirs=make_parent_dirs, args=args)
  File "C:\Users\henfri\AppData\Roaming\Python\Python37\site-packages\borg\remote.py", line 567, in __init__
    b'client_version': BORG_VERSION,
  File "C:\Users\henfri\AppData\Roaming\Python\Python37\site-packages\borg\remote.py", line 704, in call
    for resp in self.call_many(cmd, [args], **kw):
  File "C:\Users\henfri\AppData\Roaming\Python\Python37\site-packages\borg\remote.py", line 799, in call_many
    r, w, x = select.select(self.r_fds, w_fds, self.x_fds, 1)
OSError: [WinError 10038] Ein Vorgang bezog sich auf ein Objekt, das kein Socket ist
```

I do not understand, why line 148 in archiver.py is called.  This should be called if the repo is ssh, which is not the case.

I am using the *.whl, with all os.set_blocking() commented out in remote.py. otherwise it crashes even earlier.

Greetings,
Hendrik

**Comment by jrast** [comment]
Please use the `file` protocol explicitly if you are using an absolute path for the repo (as mentioned in the Windows readme):

`borg init --encryption=repokey "file://C:\Users\henfri\Desktop\temp\Borg\"`

This is a bug and should be fixed in a future version, but currently you have to use this workaround.

**Comment by jrast** [comment]
> I do not understand, why line 148 in archiver.py is called. This should be called if the repo is ssh, which is not the case.
> 
> I am using the *.whl, with all os.set_blocking() commented out in remote.py. otherwise it crashes even earlier.

The problem is that a path which starts with a drive letter (`C:\`) is currently interpreted as remote repository, which is wrong. The fix you applied only fixes this problem only at a single place but not in the whole program.

**Comment by jrast** [comment]
See also #4732

**Comment by henfri** [comment]
Hello,

I understand. The workaround that you suggest does not work though:

`OSError: [WinError 123] Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch: 'C:\\Users\\henfri\\AppData\\Roaming\\Python\\Python37\\Scripts\\file:\\C:\\Users\\henfri\Desktop\\temp\\Borg"'`

`C:\\Users\\henfri\\AppData\\Roaming\\Python\\Python37\\Scripts\\`

is the current working directory (and the one the borg.exe is in).

Note, that file:\\ is placed in between that path and the target path I specified.
It seems, that the path was interpreted as a relative path.

Greetings,
Hendrik

**Comment by jrast** [comment]
You use backward slashes `\\` instead of forward slashes `//`.

**Comment by henfri** [comment]
Hello,

that is what I tried. I did use \\. The change to \\ was done by borg, not my me. 
I am using the whl.

With the executable, I get:

```
c:\Program Files\Borg>.\borg.exe init --encryption=repokey "file://C:\Users\Hendrik Friedel\Desktop\temp\Borg\"
Local Exception
Traceback (most recent call last):
  File "borg\archiver.py", line 4434, in main
  File "borg\archiver.py", line 4366, in run
  File "borg\archiver.py", line 152, in wrapper
  File "borg\repository.py", line 188, in __enter__
  File "borg\repository.py", line 263, in create
  File "borg\repository.py", line 238, in check_can_create_repository
OSError: [WinError 123] Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch: 'c:\\Program Files\\Borg\\file:\\C:\\Users\\Hendrik Friedel\\Desktop\\temp\\Borg"'

Borg: 1.2.0a7.dev96+gd31afcb2  Python: CPython 3.7.4 msgpack: 0.6.1
PID: 15316  CWD: c:\Program Files\Borg
sys.argv: ['.\\borg.exe', 'init', '--encryption=repokey', 'file://C:\\Users\\hen fri\\Desktop\\temp\\Borg"']
SSH_ORIGINAL_COMMAND: None
```

Greetings,
Hendrik

**Comment by jrast** [comment]
Ah sorry, you need to use forward slashes everywhere: `borg init --encryption=repokey file://D:/Temp/BorgRepo`

**Comment by henfri** [comment]
That way, it worked, thanks.
I suggest though, to make backward slashes possible.

Greetings,
Hendrik

**Comment by jrast** [comment]
I think this can be closed as dupplicate of #4732

**Comment by henfri** [comment]
Yes. 
Working around it, I was able to do my first successful (local) Backup :-)
Thanks!
